src/bodies.cpp

```diff
@@ -579,76 +579,72 @@ void bodies::Box::computeBoundingCylinder(BoundingCylinder &cylinder) const
 
 bool bodies::Box::intersectsRay(const Eigen::Vector3d& origin, const Eigen::Vector3d& dir, EigenSTL::vector_Vector3d *intersections, unsigned int count) const
 {
-  if (detail::distanceSQR(center_, origin, dir) > radius2_) return false;
-
-  double t_near = -std::numeric_limits<double>::infinity();
-  double t_far  = std::numeric_limits<double>::infinity();
 
-  for (int i = 0; i < 3; i++)
+  //Brian Smits. Efficient bounding box intersection. Ray tracing news 15(1), 2002
+  float tmin, tmax, tymin, tymax, tzmin, tzmax;
+  float divx, divy, divz;
+  divx = 1 / dir.x();
+  if(divx >= 0)
   {
-    const Eigen::Vector3d &vN = i == 0 ? normalL_ : (i == 1 ? normalW_ : normalH_);
-    double dp = vN.dot(dir);
-
-    if (fabs(dp) > detail::ZERO)
-    {
-      double t1 = vN.dot(corner1_ - origin) / dp;
-      double t2 = vN.dot(corner2_ - origin) / dp;
+    tmin = (corner1_.x() - origin.x()) * divx;
+    tmax = (corner2_.x() - origin.x()) * divx;
+  } else
+  {
+    tmax = (corner1_.x() - origin.x()) * divx;
+    tmin = (corner2_.x() - origin.x()) * divx;
+  }
 
-      if (t1 > t2)
-        std::swap(t1, t2);
+  divy = 1 / dir.y();
```

Comment:
divide by zero

Reply:
division of a finite nonzero float by 0 is well-defined and results in +infinity (if the value was >zero) or -infinity (if the value was less than zero).